### PR TITLE
Feature/dark mode

### PR DESCRIPTION
**Pull request by mr-ryan12**
What (if any) features are you implementing?
- Added dark mode functionality

What (if anything) did you refactor?
- Refactored SASS to pass in dark mode elements

Were there any issues that arose?
- Not that weren't resolved

Is there anything that you need from your teammate?
- PR review and comments

Any other comments, questions, or concerns?
- Not at the moment

### CODE REVIEW

**Review by tonycarpenter21 — APPROVED** [review]
Fantastic work! I love how you concatenated the classNames to make the logic work.

File: src/404/404.tsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
-const ErrorHandling: React.FC<ErrorHandlingProps> = ({ message }) => {
+const ErrorHandling: React.FC<ErrorHandlingProps> = ({ message, darkMode }) => {
   return (
     <section className="error">
-      <img className='error-pic' src="https://media2.giphy.com/media/rYEAkYihZsyWs/giphy.gif?cid=ecf05e47mxw0p2ev880va40mrexkbc8mm0gvmprvp8elek15&rid=giphy.gif&ct=g" alt="Bob Ross 404 error"/>
+      <img className="error-pic" src="https://media2.giphy.com/media/rYEAkYihZsyWs/giphy.gif?cid=ecf05e47mxw0p2ev880va40mrexkbc8mm0gvmprvp8elek15&rid=giphy.gif&ct=g" alt="Bob Ross 404 error"/>
       <h1 className="error-message">{message}</h1>
-      <NavLink to="/" className='return-to-main'>Back To Main Page</NavLink>
+      <NavLink to="/" className={"return-to-main" + darkMode}>Back To Main Page</NavLink>
```
Comment (StevieAmb) [inline src/404/404.tsx:12]: I love the appending of the prop that changes the CSS rules SO much!

File: src/App.tsx [hunk 2]
```diff
@@ -13,29 +13,45 @@ class App extends Component<{}, AppState> {
 
     this.state = {
       categories: ['photography', 'watercolor', 'painting', 'oil painting', 'sculpture', 'pencil', 'pastel', 'chalk'],
-      darkMode: '',
+      darkMode: localStorage.length === 0 ? '' : localStorage.getItem('dark-mode'),
       isDarkModeOn: false
     }
   }
 
+  componentDidMount = () => {
```
Comment (StevieAmb) [inline src/App.tsx:21]: I'm really stoked about the use of local storage to have the styling persists!

File: src/App.tsx [hunk 3]
```diff
@@ -13,29 +13,45 @@ class App extends Component<{}, AppState> {
...
+      document.body.style.color = '#000'
       this.setState({
         isDarkModeOn: false,
-        darkMode: ''
+        darkMode: localStorage.getItem('dark-mode')
       })
     } else {
+      localStorage.setItem('dark-mode', '-dark')
+      document.body.style.backgroundColor = '#000'
+      document.body.style.color = '#FFF'
```
Comment (StevieAmb) [inline src/App.tsx:51]: LocalStorage and if/else statements, very awesome!

**Review by StevieAmb — APPROVED** [review]

**Pull request merged**